### ISSUE TITLE
Matter Beamer no longer beams Living matter

### ISSUE THREAD

**Issue by ZyorTaelon** [McJty/RFTools#1943]
Issue description: Matter Beamer is powered and has redstone signal. It won't send any living matter to the connected spawner. When I replace the living matter with soul sand or a nether star it's sends just fine.

Steps to reproduce: 
![image](https://user-images.githubusercontent.com/1486566/56963750-f947bb80-6b59-11e9-954b-46cb96f35efb.png)
Place Matter Beamer, connect to Spawner with Wither syringe. Power the Matter Beamer and place a lever on it in the 'on' position.
Add a nether star into the beamer. Watch it disappear and end up in the spawner.
Add carrots, potatoes, leaves or raw beef to the beamer. See it stay there for all eternity :)

Versions:
* Minecraft: 1.12.1
* Forge: forge-1.12.2-14.23.5.2808-universal
* CompatLayer (only if on Minecraft 1.10 or 1.11):
* McJtyLib: mcjtylib-1.12-3.5.0
* RFTools: rftools-1.12-7.70

Possibly incompatible mods and versions: 

Relevant logs, if any: 
There are no new lines on the log or server console when I toggle the redstone signal.

**Comment by ZyorTaelon** [comment]
[rftools.cfg.zip](https://github.com/McJtyMods/RFTools/files/3131372/rftools.cfg.zip)

**Comment by Famous5000** [comment]
Intentional. Withers have different spawn requirements. Please check a wiki if you want more info.

**Comment by ZyorTaelon** [comment]
Did that change in the last release?

**Comment by Famous5000** [comment]
Nope. They cannot accept normal living matter. The spawn tables should be the same unless you have modified them.

**Comment by McJty** [comment]
Hmm, that's not true. Withers accept normal living matter like everything else. There is definitely a bug here

**Comment by Famous5000** [comment]
Oh, I thought... Did you check the spawner GUI @ZyorTaelon? It may not be letting biotic matter through because of it being full

**Comment by McJty** [comment]
The problem is probably related to me changing how rftools reads the config files. I think this messes up the internal table. I will fix this

**Comment by ZyorTaelon** [comment]
Probably because of the empty line in the living matter configs

**Comment by LostDGod** [comment]
@ZyorTaelon You can manually fix this issue in your client by removing rftools.cfg and restart the game to let it create a new one. If you had done changes in file you will have to add them again.

**Comment by McJty** [comment]
This also has to be done on the server. Not sure what went wrong though

**Comment by LostDGod** [comment]
This issue happened to me after updating RFTools. Unfortunately I do not have my original config file before the update, but I do have the bugged file that was constructed by the updated mod and the new one that was created once I deleted the bugged one. 
The only changes I had done (prior to the update and after deleting the bugged file) was to add syringes for thermalfoundation mobs and some vanilla mobs which did not have any such as the elder guardian and evoker.
Just in case you might want to look into it, here are my files:
[bugged config](https://mega.nz/#!coZS2YqZ!eWyBW7JpwcOclzsenWu1OjAd0jAgQu8HAhYpQv_zFL4)
[working config](https://mega.nz/#!lwBSCSrA!KmCHSUuk9oXpie52qLytjMiLB7Ph4XCX74VUcRibdDQ)

**Comment by VADemon** [comment]
@McJty bump. After comparing files before and after it appears the only change to config are the types of stored variables throughout the file. Mostly `D` types were converted to `S` type etc. I didn't spot any other changes. Still beyond me why it affects the Living Matter configuration while everything else works. Does it default to empty if the mod fails to read the config?

Broken and regenerated+adapted config here: https://gist.github.com/VADemon/7844e8bed26b9d9ab581dad796034942

**Comment by Didiervolk00** [comment]
Same problem here, my spawner won't accept the wheat (living matter) I want to use for the wither spawn process...

**Comment by VADemon** [comment]
@Didiervolk00 Two people already posted the working configs above: it's a config incompatibility issue. Or just delete the existing one.

**Comment by Didiervolk00** [comment]
I'm not into this kind of playing minecraft, but where do I change this config?

**Comment by JTBlum** [comment]
This worked for me. I was playing Stone Block 2, and closed out before renaming "rftools.cfg" in the "rftools" folder to "rftools.cfg.bak" in case I lost data.

**Comment by laundmo** [comment]
just going to add this here, this is the new file for stoneblock2 with all the values adjusted to match the original stoneblock2 config

[rftools.cfg.txt](https://github.com/McJtyMods/RFTools/files/5113618/rftools.cfg.txt)

**Comment by monocleduck** [comment]
Hi, im also experiencing this issue in Stoneblock2 and deleting the config, closing the game and adding the new one doesnt seem to do the trick. Is there another way to fix it?

**Comment by kentonc04** [comment]
> Hi, im also experiencing this issue in Stoneblock2 and deleting the config, closing the game and adding the new one doesnt seem to do the trick. Is there another way to fix it?

same

**Comment by Raidobw2** [comment]
A bunch of players are having the same issue on Stoneblock 2 and coming on the FTB Discord, so i'll post another workaround in case @laundmo 's workaround did not work for them or that deleting the config didn't work for them as well (although that's weird it didn't work for either). <ins>**That said, his workaround did work for me and it looks like it was a config problem, please try that first. Deleting the config also worked.**<ins>

Versions for anyone wondering:
- Minecraft: 1.12.2
- Forge: 1.12.2-14.23.5.2847
- McJtyLib: mcjtylib-1.12-3.5.4
- RFTools: rftools-1.12-7.73
All of this ^ obtainable from Stoneblock 2 version 1.21.1, fresh install. 

Being honest, the players will only need it for a few mobs, so the "fix" is not hard to apply and not needed for a lot of mobs. I am basing my explanation off Stoneblock 2 especially and with common code logic, version is above.
You need to be resourceful, ~~i won't go too much in detail~~. 

First, the config is in `/config/rftools/rftools.cfg`. You'll have some sections under `mobspawnamounts` and that's where we'll fix it. In this example, we will fix the "Zombie" mob from Vanilla and add wheat (minecraft:wheat) as the living item needed for its spawn.

In the section at the left of the image, you'll see letters and it basically comes back to this, from what i understand:
The I stands for "**I**-tem" (item). The B stands for "**B**-lock" (block). The L stands for "Living Matter", since they have a whole list at the start of the config file. We will then focus on the "Red Box" in that left section of the picture, because that's the section for the third ingredient for the spawner.

If i want to use wheat for my living matter, i will replace the "L" on the first yellow line with an "I" because wheat is an "**I**.tem" and not a "**B**.lock". Then i will add "minecraft:wheat" on the second yellow line (the empty one). It will look like in picture 2, right.

![image](https://user-images.githubusercontent.com/17010632/130927248-c2a08664-2f9e-4348-8675-e4b19568151f.png)

The "20.0" in the picture on both sides means "20 wheat" in our example, or at least i think it means that (haven't fully tested that part). Feel free to lower it based on the section `livingmatter` in the config, for example wheat in reality looks like:
```
S:living.16 <
        I
        minecraft:wheat
        1.1
     >
```
So this means 1 wheat gives 1.1 living matter. A zombie needs 20 living matter before modification. So we would divide 20 by 1.1 and we obtain roundly 18 wheat. Changing the "20.0" to a "18.0" would then make this spawn balanced. Although, you do you, and since this is bugged for you for some reason, just lower it more since you're handicapped to using one item (e.g.:the wheat) for this, if you feel like doing so.

Then save the config file and boot back in the game and the last beamer will accept the "**I**.tem" you put on yellow line 2, the "minecraft:wheat", of course when receiving a redstone signal and power. For pumpkins (minecraft:pumpkin), that would be a "**B**-lock" so you'd put a 'B', for wheat seeds (minecraft:wheat_seeds) they are "**I**.tems" and so on, you get the idea. Not the best workaround but hey, that should work, it's all that matters sometimes.

For Stoneblock 2, to find the name of a living item, you have access to CraftTweaker commands in-game. Hold, say, dandelions in your hand and type in `/ct hand` in chat then Enter and the name will show up.

Wheat seeds can be planted though, what about them actually? Are they a block or item? Well, that i can't say much, but planted they're listed as this:

![image](https://user-images.githubusercontent.com/17010632/130929656-af23cb27-20fa-4f88-a84e-640b072d41e3.png)

So just go with your gut feeling if you think that's a block or item and at worst you'll try again. Hopefully information in this post is accurate, sorry if i missed something, @McJty . Mod seems to be behaving fine after laundmo's config fix. Mod is also behaving fine after a config deletion and booting up the game again for the versions listed above, i just tested both cases on a fresh install. If this is sufficient for you, this could be closed as the mod seems to be behaving properly, although, no idea why some users can't get it to work like me and others did in this post. Maybe they're poking at the wrong file or something. Looks like the old RFTools config in Stoneblock 2 stayed for some time, hitting this thread. Was going to delete my "workaround" after i saw it working for both cases, but i'll leave it just in case it works better for the 2 users waiting above me, sorry for the brick of text!

-R3GEN
